plotter/heat_test.go

```diff
@@ -58,6 +60,16 @@ func ExampleHeatMap() {
 
 	p.Add(h)
 
+	// Create a legend.
+	rng := h.Max - h.Min
+	thumbs := PaletteThumbnailers(plte)
+	for i := len(thumbs) - 1; i >= 0; i-- {
+		t := thumbs[i]
+		fmin := float64(i) / float64(len(thumbs))
+		fmax := float64(i+1) / float64(len(thumbs))
+		p.Legend.Add(fmt.Sprintf("%.2g - %.2g", h.Min+fmin*rng, h.Min+fmax*rng), t)
+	}
+
```

Comment:
This for loop seems like a good candidate for incorporation into the API as `BuildThumbnail` and (if so, additionally) incorporated into an `AddThumbnail` method.